src/hooks/useMessage.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState
+} from 'react';
+import { useLocation } from 'react-router-dom';
+
+const MessageContext = createContext<
+  | {
+      message: string | undefined;
+      showMessage: (message: string) => void;
+      showMessageOnNextPage: (message: string) => void;
+    }
+  | undefined
+>(undefined);
+
+// MessageProvider manages the state necessary for child components to
+// use the useMessage hook.
+const MessageProvider = ({ children }: { children: ReactNode }) => {
+  const [queuedMessage, showMessageOnNextPage] = useState<string>();
```

Comment:
Additionally, can we have these values (`queuedMessage`, `message`) default to empty string rather than `undefined`? I'm not sure if there's a reason to have `undefined`. It'll help tighten Typescript too.

Reply:
I agree. I intended to keep the `set` names in this component but caught these during some renaming.